bitcoin/src/blockdata/script/witness_program.rs

```diff
@@ -164,3 +164,45 @@ impl std::error::Error for Error {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn witness_program_is_too_short() {
+        let bytes = [0x00; MIN_SIZE - 1]; // Arbitrary bytes
+        assert!(WitnessProgram::new(WitnessVersion::V15, &bytes).is_err()); // Arbitrary version
+    }
+
+    #[test]
+    fn witness_program_is_too_long() {
+        let bytes = [0x00; MAX_SIZE + 1]; // Arbitrary bytes
+        assert!(WitnessProgram::new(WitnessVersion::V15, &bytes).is_err()); // Arbitrary version
+    }
+
+    #[test]
+    fn valid_v0_witness_programs() {
+        let bytes = [0x00; MAX_SIZE]; // Arbitrary bytes
+
+        for size in MIN_SIZE..=MAX_SIZE {
+            let program = WitnessProgram::new(WitnessVersion::V0, &bytes[..size]);
+
+            if size == 20 {
+                assert!(program.expect("failed to create witness program").is_p2wpkh());
+                continue
+            }
+            if size == 32 {
+                assert!(program.expect("failed to create witness program").is_p2wsh());
+                continue
+            }
+            assert!(program.is_err());
+        }
+    }
+
+    #[test]
+    fn valid_v1_witness_programs() {
+        let bytes = [0x00; 32]; // Arbitrary bytes
```

Comment:
I find the comments confusing.  Probably could do without `Arbitrary bytes` comment?

Reply:
I thought it might clear up any confusion on if we actually care about the contents of the witness programs being created. Since we don't in this case, i decided to add the comment. I'll rename `bytes` to `arbitrary_bytes`, maybe that will be less confusing